### PR TITLE
[RG] Atomically refresh RG sets.

### PR DESCRIPTION
**Pull request by kobolog**
@antmat @3Hren @noxiouz

### CODE REVIEW

File: src/service/locator.cpp [hunk 1]
```diff
@@ -495,43 +495,35 @@ locator_t::on_connect(const std::string& uuid) -> streamed<results::connect> {
...
+            continue;
+        } else {
+            COCAINE_LOG_INFO(m_log, "updating routing group");
+        }
 
-        if(lb != ub) {
-            mapping.insert({*it, continuum_t(std::move(group_log), lb->second)});
+        try {
+            clone.insert({*it, {
+                std::make_unique<logging::log_t>(*m_log, attribute::set_t()),
```
Comment (kobolog) [inline src/service/locator.cpp]: @3Hren: is there some sort of `clone()` method?

Comment (3Hren) [inline src/service/locator.cpp]: Nope.

**Comment by 3Hren** [comment]
:+1:

**Comment by noxiouz** [comment]
+1

**Comment by 3Hren** [comment]
But I'd rather add some comments, personally it wasn't trivial for me to understand how it works.

**Pull request merged**